depsync/syncer.go

```diff
@@ -0,0 +1,149 @@
+package depsync
+
+import (
+	"errors"
+	"fmt"
+	"github.com/go-kit/kit/log"
+	"github.com/micromdm/dep"
+	"time"
+)
+
+// Syncer is responsible for fetching devices from DEP, keeping them in sync, and storing cursor information about the
+// current fetch operation.
+type Syncer interface {
+	Start(deviceChan chan dep.Device)
+	Pause()
+	Resume()
+	Fetch(deviceChan chan dep.Device) (bool, error)
+}
+
+// Cursor represents information about the current fetch operation. It is required to fetch multiple "pages" of devices.
+type Cursor struct {
+	Value   string
+	Created *time.Time // >7 days cursor is invalid.
+}
+
+type syncer struct {
+	logger               log.Logger
+	client               dep.Client
+	tickerChan           <-chan time.Time
+	tickerChanDisabled   <-chan time.Time
+	doneChan             <-chan struct{}
+	Cursor               *Cursor
+	InitialFetchComplete bool
+	errorCount           int
+}
+
+func NewSyncer(client dep.Client, logger log.Logger, tickerChan <-chan time.Time) Syncer {
+	return &syncer{
+		logger:     logger,
+		client:     client,
+		tickerChan: tickerChan,
+		errorCount: 0,
+	}
+}
+
+// Fetch fetches a list of DEP devices in batches. The first return value indicates that there are more devices to fetch.
+func (s *syncer) Fetch(deviceChan chan dep.Device) (bool, error) {
```

Comment:
`deviceChan` should be renamed to `device` and the channel input needs a direction specified. Is this a send or receive channel?